proctor-common/src/main/java/com/indeed/proctor/common/ForceGroupsOptionsStrings.java

```diff
@@ -62,26 +71,105 @@ public static ForceGroupsOptions parseForceGroupsString(@Nullable final String f
             }
             //  bucketValueStart should now be the index of the minus sign or the first digit in a run of digits going to the end of the word
             final String testName = piece.substring(0, bucketValueStart).trim();
-            final String bucketValueStr = piece.substring(bucketValueStart);
+            final String bucketValueStr = piece.substring(bucketValueStart).trim();
             try {
                 final Integer bucketValue = Integer.valueOf(bucketValueStr);
                 builder.putForceGroup(testName, bucketValue);
+                if (bucketAndPayloadValuesStr.length == FORCE_PARAMETER_MAX_SIZE) {
+                    final Payload payloadValue = parseForcePayloadString(bucketAndPayloadValuesStr[FORCE_PARAMETER_PAYLOAD_IDX]);
+                    if (payloadValue != null){
+                        builder.putForcePayload(testName, payloadValue);
+                    }
+                }
             } catch (final NumberFormatException e) {
-                LOGGER.error("Unable to parse bucket value " + bucketValueStr + " as integer", e);
+                LOGGER.error("Unable to parse bucket value " + bucketAndPayloadValuesStr[FORCE_PARAMETER_BUCKET_IDX] + " as integer", e);
             }
         }
         return builder.build();
     }
 
+    @Nullable
+    public static Payload parseForcePayloadString(final String payloadString )
```

Comment:
Could you document the expected format of the encoded payload in URL parameter? It seems it's not JSON. We have to carefully consider the encoded payload format.

Reply:
Thanks.
We'll introduce "JSON" payload type (currently the design is in progress). I'm afraid that we cannot write nested payload values in this format.

Reply:
To handle JSON format in the future, I can update the parsing of the force group string to rather than just split on "," to instead iterate through the string and split the substring when a "," is found and not within a payload. e.g. (test1,test2;map:{key1:value1, key2:value2},test3;JSON{...,...} would be split into 
["test1",    "test2;map:{key1:value1, key2:value2}",    "test3;JSON{...,...}"])